### PR TITLE
Salambashir/addDFSv2Swagger

### PR DESCRIPTION
**Pull request by salambashir**
### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [X] new service onboarding
      - [x] new API version
      - [x] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  The service is in Public Preview at the moment.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  ASAP
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.
  We still don't have any SDKs

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version). Note that this doesn't apply if you are trying to merge a PR that was previously in the private repository.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/overview_breakingchanges).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](http://aka.ms/aprwiki))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by kazrael2119** [comment]
@salambashir ,please fix LintDiff, Avocado and ModelValidation error

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (ms-zhenhua) [inline specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:6]: >2022-12-01-preview [](http://example.com/codeflow?start=16&length=18)

Why use such old date? The latest date is 2023-05-01-preview:
[https://github.com/Azure/azure-rest-api-specs/tree/main/specification/security/resource-manager/Microsoft.Security/preview/2023-05-01-preview](https://github.com/Azure/azure-rest-api-specs/tree/main/specification/security/resource-manager/Microsoft.Security/preview/2023-05-01-preview)

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (ms-zhenhua) [inline specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:37]: >"/{resourceId}/providers/Microsoft.Security/defenderForStorageSettings/{settingName}" [](http://example.com/codeflow?start=4&length=85)

Please have a conversation with the PAS team for APIs outside the subscription context. You can send an email to prasnal@microsoft.com to kick start the review

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (salambashir) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:6]: The API was approved before with this API version before but wasn't merged. It is already public...

**Comment by salambashir** [comment]
@ms-zhenhua, I see that one of the LintDiff errors is the following: 
"The top-level resource 'DefenderForStorageSetting' does not have list by subscription operation, please add it.
Location: [Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json#L127](https://github.com/Azure/azure-rest-api-specs/blob/c72bc7c568005b4de7ac2e9c01c3c97b2e8c3d0b/specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json#L127)"

However, DefenderForStorageSettings is not a top-level resource, and we don't want to enable list operation on the subscription for it. Do you know how we can bypass this?

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (salambashir) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:6]: The PR that was approved was on the private repo.

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (yoavfr) [inline specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:210]: Perhaps "overall success/failure"?

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (yoavfr) [inline specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:208]: Why is the capitalization of this OperationStatus different from the others?

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (yoavfr) [inline specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:239]: Should "Settings" be capitalized?

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (yoavfr) [inline specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:250]: capitalization of Settings?

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (salambashir) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:114]: This is the "default" response, not the 201 code.

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (salambashir) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:182]: all should be lowercase. I will update the operation status

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (salambashir) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:210]: I don't think it's right to use overall here. There are two operationStatus fields in in the response of the PUT. One for MalwareScanning and one for DataSensitivity. See the example of the PUT request in the examples folder.

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (salambashir) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:208]: The convention is that definitions are capitalized, but the properties inside the definitions are not. This is what I did in this file.
This is relevant for your other comments as well...

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (salambashir) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:239]: See previous reply

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (salambashir) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:250]: See previous reply

**Comment by kazrael2119** [comment]
@salambashir ,if you finish Arm change requests, you should remove ```ArmChangesRequested``` label and re-add ```WaitForARMFeedback``` label

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (zizw123) [inline specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:51]: >v1 [](http://example.com/codeflow?start=37&length=2)

Consider use latest v5 when possible

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (zizw123) [inline specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:38]: >get [](http://example.com/codeflow?start=7&length=3)

Only get and put? Can you add more context here? Please see [https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review)

Comment (salambashir) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:38]: Hi, yes this is a proxy only resource and we need to implement Get and Put only.
This is a capability that can be turned on/off on a storage account. Put is used to define the desired state (on/off) and Get will show the current state.
These are the only operation we need.

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (salambashir) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:51]: These are the common types defined for the "security" resource-manager.
Security has only v1 defined for it's resource-manager

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (ms-henglu) [inline specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:212]: Consider model it as an enums if the property has a finite set of values?

refs: https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi015-avoid-modelling-a-property-as-string-type-when-it-should-be-an-enums

**Review by ms-henglu — CHANGES_REQUESTED** [review]

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (ms-henglu) [inline specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:143]: Consider using enum instead:
https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md#why-the-rule-is-important

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (ms-henglu) [inline specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:155]: Consider using enum instead:
https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md#why-the-rule-is-important

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (ms-henglu) [inline specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:181]: Consider using enum instead:
https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md#why-the-rule-is-important

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (ms-henglu) [inline specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:196]: Consider using enum instead:
https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md#why-the-rule-is-important

**Review by ms-henglu — CHANGES_REQUESTED** [review]

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (salambashir) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:143]: I am not sure what are the advantages of enum here. After all, defining this as boolean limits the possible values to True and False...

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (salambashir) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:196]: See reply in the earlier comment

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (salambashir) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:155]: See reply on the previous comment

File: specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json
Comment (salambashir) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json:212]: The possible values of the code are not defined as a known predefined list yet.
Defining these value may come later when we work on a non-preview version of the API.
For now, we don't want to limit the values passed here.

**Review by ms-henglu — APPROVED** [review]

**Comment by salambashir** [comment]
@kazrael2119 I am unable to merge the PR. Can you please merge it?

**Comment by kazrael2119** [comment]
@tadelesh ,please help review go sdk breaking change, thanks

**Review by kazrael2119 — APPROVED** [review]

**Comment by kazrael2119** [comment]
> @kazrael2119 I am unable to merge the PR. Can you please merge it?

@salambashir ,do you want to merge this pr now? if not, please let me know when you want to merge it, thanks

**Comment by yoavfr** [comment]
> > @kazrael2119 I am unable to merge the PR. Can you please merge it?
> 
> @salambashir ,do you want to merge this pr now? if not, please let me know when you want to merge it, thanks

@kazrael2119 - Yes. We would like to have this merged. 
As a side note, it would greatly ease communication if we could communicate directly over Teams. Unfortunately, we have been unable to figure out your MS alias from you handle in github.

**Pull request merged**
